cou/cli.py

```diff
@@ -12,140 +12,164 @@
-    :type args: Any
-    :return: Arguments parsed to the cli execution.
-    :rtype: argparse.Namespace
+    - 'ERROR': Only errors will be logged.
+    - 'WARNING': Both errors and warnings will be logged.
+    - 'INFO': Errors, warnings, and general information will be logged.
+    - 'DEBUG': Detailed debugging information will be logged.
+    - 'NOTSET': Maximum verbosity where everything will be logged.
     """
-    parser = argparse.ArgumentParser(
-        description="Charmed OpenStack Upgrader(cou) is an application to upgrade Charmed "
-        "OpenStack. Application identifies the lowest OpenStack version on the components and "
-        "upgrade to the next version.",
-        formatter_class=argparse.RawDescriptionHelpFormatter,
-        exit_on_error=False,
-        add_help=False,
-    )
-    parser.add_argument(
-        "--run",
-        help="Use this flag to run the upgrade, otherwise just print out the upgrade steps.",
-        action="store_true",
-        default=False,
-    )
-    parser.add_argument(
-        "--log-level",
-        default="INFO",
-        dest="loglevel",
-        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
-        type=str.upper,
-        help="Set the logging level. Defaults to INFO. This only affects stdout. The logfile "
-        "will be always DEBUG. The file location is in COU_DATA/log. You can set the "
-        "COU_DATA via environment variable. However it needs to be plugged to snap.",
-    )
-    parser.add_argument(
-        "--model",
-        default=None,
-        dest="model_name",
-        help="Set the model to operate on. If not set it gets the model name in this order:\n"
-        "  1 - Environment variable JUJU_MODEL,"
-        "  2 - Environment variable MODEL_NAME,"
-        "  3 - Current active juju model",
-    )
-    parser.add_argument(
-        "--interactive",
-        help="Run upgrade with prompt.",
-        action=argparse.BooleanOptionalAction,
-        default=True,
-    )
-    parser.add_argument(
-        "-h",
-        "--help",
-        action="help",
-        default=argparse.SUPPRESS,
-        help="Show this help message and exit.",
-    )
-
-    return parser.parse_args(args)
-
-
-def setup_logging(log_level: str = "INFO") -> None:
-    """Do setup for logging.
-
-    :returns: Nothing: This function is executed for its side effect
-    :rtype: None
+
+    ERROR = 0
+    WARNING = 1
+    INFO = 2
+    DEBUG = 3
+    NOTSET = 4
+
+    @classmethod
+    def _missing_(cls, value: object) -> Enum:
+        """Return maximum verbosity for value larger than 4.
+
+        :param value: value to get enum member
+        :type value: object
+        :return: return a member of VerbosityLevel
+        :rtype: Enum
+        :raises ValueError: Invalid value input.
+        """
+        if isinstance(value, int) and value > 4:
+            return cls.NOTSET
+        raise ValueError(f"{value} is not a valid member of VerbosityLevel.")
```

Comment:
If I'm correct this can be raised by `get_log_level` in `entrypoint`, and it is not catch properly. We will reach
```python
logger.error("Unexpected error occurred")
logger.exception(exc)
sys.exit(2)
```
What is not correct.

Reply:
In theory, yes, `VerbosityLevel(-1)` would cause an exception, but as a user you can only instantiate `VerbosityLevel(0)` and up.

Reply:
Yes. If the exception is ever raise, that means something unexpected happened and I think `exit(2)` is appropriate for it

Reply:
This `ValueError` confused me, but you are right it could not happen from any CLI call.